### DISCUSSION TITLE
Copy current URL with a keyboard shortcut

### DISCUSSION THREAD

**Discussion by cizeon** [zen-browser/desktop#2631] [Ideas]
One of the feature I love with the Arc browser is the ability to quickly copy current URL with a simple shortcut: CTRL+SHIFT+C or CMD+SHIFT+C.

I'd love to have the same shortcut in Zen Browser.

Thank you for the great work!

**Comment by rubaboo** [comment]
I'm ok with 2 shortcuts that are used almost universally. Less things to remember.
<kbd>Alt</kbd>+<kbd>D</kbd> Focus the adddres bar
<kbd>Ctrl</kbd>+<kbd>C</kbd> Copy

**Reply by cizeon** [reply]
Thank you for your response.

Having it as an option would be great. You can still use your workflow with two shortcuts, while I can also do mine with only one ;-).

**Reply by mmsalois** [reply]
`Ctrl`+ `L` also works for focusing the address bar

**Reply by rubaboo** [reply]
I know, multiple options exist. No point wanting to make one browser exactly like the other one.

**Reply by valk-ryx** [reply]
> Thank you for your response.
> 
> Having it as an option would be great. You can still use your workflow with two shortcuts, while I can also do mine with only one ;-).

At the moment I'm using the "Shortkeys" addon which allows you to do this easily, I've also found it helpfully blocks the default ctrl+shift+c shortcut for the inspect element and only leaves it with the ctrl+shift+i which I think is better anyway.

**Comment by lumpysimon** [comment]
I'd love to see this too. I've lost count of how many times I've pressed `Cmd`+`Shift`+`C` in Zen and it's opened up dev tools!

**Reply by bpavuk** [reply]
same, with the recent update, my only "Arc user pain point" was `Ctrl/cmd+shift+C`

**Reply by ignoxx** [reply]
same here!

**Comment by Dayton-IX** [comment]
I'd also love to see this. Coming from Arc and I keep hitting Cmd + Shift + C and opening dev tools instead haha.

**Comment by uncenter** [comment]
Looks like this feature might have been added in a recent update? Ctrl+shift+c is now copying the URL for me!

**Comment by cizeon** [comment]
Thank you for implementing it!

**Comment by l4kr** [comment]
But I do want to open dev tools.... how do I get the shortcut back?

**Reply by uncenter** [reply]
Ctrl+shift+l is the shortcut for that now.

**Reply by YussufSassi** [reply]
ctrl+shift+I is not the same shortcut as ctrl+shift+C. I can't find any way to restore that functionality.

**Reply by piqusy** [reply]
Same here, as a developer I need this shortcut. The other default shortcut was option+cmd+c, can we have that implemented or at least have it configurable in the Keyboard Shortcuts

**Comment by oolonc** [comment]
That shortcut specifically activated  [inspect element](https://firefox-source-docs.mozilla.org/devtools-user/keyboard_shortcuts/index.html#page-inspector). Is there a way to change this back? I do wish the the devs would stop messing with the keyboard shortcuts; aside from the constantly changing keybindings Zen has been great.

**Reply by syscat29** [reply]
> That shortcut specifically activated  [inspect element](https://firefox-source-docs.mozilla.org/devtools-user/keyboard_shortcuts/index.html#page-inspector). Is there a way to change this back? I do wish the the devs would stop messing with the keyboard shortcuts; aside from the constantly changing keybindings Zen has been great. 

What version of zen are you on?

**Reply by oolonc** [reply]
Hello, 

package details:
```shell
$ pacman -Qi zen-browser-bin
Name            : zen-browser-bin
Version         : 1.0.2.b.5-1
Description     : Performance oriented Firefox-based web browser
Architecture    : x86_64
URL             : https://github.com/zen-browser/desktop
Licenses        : MPL-2.0
Groups          : None
Provides        : zen-browser=1.0.2.b.5
Depends On      : gtk3  libxt  mime-types  dbus-glib  nss  ttf-font  systemd
Optional Deps   : ffmpeg: H264/AAC/MP3 decoding [installed]
                  networkmanager: Location detection via available WiFi networks [installed]
                  libnotify: Notification integration [installed]
                  pulse-native-provider: Audio support [installed]
                  speech-dispatcher: Text-to-Speech
                  hunspell-en_US: Spell checking, American English
Required By     : None
Optional For    : None
Conflicts With  : zen-browser
Replaces        : None
Installed Size  : 292.72 MiB
Packager        : Unknown Packager
Build Date      : Wed 25 Dec 2024 11:16:39 PM MST
Install Date    : Wed 25 Dec 2024 11:16:48 PM MST
Install Reason  : Explicitly installed
Install Script  : No
Validated By    : None
```

**Reply by uncenter** [reply]
Try restarting your browser. I also noticed the shortcut toggling inspect element for a build or so, updated/restarted again and it was back to copying the URL as expected.

**Reply by syscat29** [reply]
> Try restarting your browser. I also noticed the shortcut toggling inspect element for a build or so, updated/restarted again and it was back to copying the URL as expected.

Hmm i believe this was fixed in b.5 but yeah restart your browser that should fix it.

**Reply by Unstate** [reply]
> That shortcut specifically activated [inspect element](https://firefox-source-docs.mozilla.org/devtools-user/keyboard_shortcuts/index.html#page-inspector). Is there a way to change this back? I do wish the the devs would stop messing with the keyboard shortcuts; aside from the constantly changing keybindings Zen has been great.

As of May 4, 2025, it may be possible to restore the "Pick an element from the page" shortcut!
The following steps worked for 
1)Remove the Ctrl + Shift + C shortcut from the "Copy URL" option in Zen Browser settings.
2)Go to the Developer Tools section in Zen Browser settings and locate the "Toggle Inspector" option.
3)Assign the Ctrl + Shift + C shortcut to this option. It should now work correctly!

**Comment by Jarpiino** [comment]
I love(d) using zen browser, but do not change browser's core features, ty

**Comment by Jarpiino** [comment]
With update 1.6b you can now bind the inspector back to CTRL + SHIFT + C.
